pkg/deployment/agent.go

```diff
@@ -32,6 +32,15 @@ func NewAgent(jaeger *v1.Jaeger) *Agent {
 
 // Get returns a Agent pod
 func (a *Agent) Get() *appsv1.DaemonSet {
+	// Check if the agent is enabled
+	if a.jaeger.Spec.Agent.Enabled == nil || !*a.jaeger.Spec.Agent.Enabled {
```

Comment:
```suggestion
	if a.jaeger.Spec.Agent.Enabled == nil || !*a.jaeger.Spec.Agent.Enabled {
```
I would enable it by default, since it is the current behavior.